docs/mesos/MesosClusterPreparation.md

```diff
@@ -0,0 +1,139 @@
+of each service on a single centos7 host.** Consult the appropriate
+documentation for each service:
+    - [etcd clustering](https://coreos.com/etcd/docs/latest/clustering.html)
+    - [zookeeper multi-server setup](https://zookeeper.apache.org/doc/r3.3.2/zookeeperAdmin.html#sc_zkMulitServerSetup)
+    - [mesos high-availability mode](http://mesos.apache.org/documentation/latest/high-availability/)
+
+    For more information on how to set up a more production-ready
+    calico-mesos cluster in such a manner, [contact us on slack][slack].
+
+- **A Note on Calico's Customizations in Mesos**: It is important
+to note that no customizations are necessary for any of the
+aformentioned services to be compatible with a Calico-Mesos Agent.
+**Adding calico to a mesos cluster only requires modifications to each Agent.**
+
+
+## Prerequisite: Add the Mesos Official Repository
+Mesos-master, zookeeper, and marathon are installed from the official Mesos repository. Add the official repository:
+
+    rpm -Uvh http://repos.mesosphere.com/el/7/noarch/RPMS/mesosphere-el-repo-7-1.noarch.rpm
+
+## ZooKeeper
+Install Mesos' datastore - ZooKeeper:
+
+```
+yum install -y mesosphere-zookeeper
+systemctl start zookeeper
+```
+
+ZooKeeper uses tcp over port 2181. If you're using a firewall, open this port:
+
+```
+sudo firewall-cmd --zone=public --add-port=2181/tcp --permanent
+sudo systemctl restart firewalld
+```
+
+## Mesos Master
+Supported Versions: `0.28` (recommended, required for Docker Containerizer),
+`v0.27`, `v0.26.0` (deprecated)
+
+With Zookeeper up and running, we can launch our mesos-master process. Run the following commands, replacing `$IP` with the cluster-accessible IP you would like Master to bind to:
+
+```
+yum install -y mesos
+echo $IP > /etc/mesos-master/hostname
+echo $IP > /etc/mesos-master/ip
+systemctl start mesos-master
+```
+
+## Marathon
+Supported Versions: `v1.0.0` (recommended), `v0.14.0` - `v0.14.2`
+
+Once mesos-master is running, we can launch Marathon:
+
+```
+sudo yum install -y marathon
+systemctl start marathon
+```
+
+Marathon listens for tcp connections on port 8080. If you're using a firewall,
+open this port:
+
+```
+sudo firewall-cmd --zone=public --add-port=8080/tcp --permanent
+sudo systemctl restart firewalld
+```
+
+## etcd
+Calico uses etcd as its data store and communication mechanism among Calico components. Install it from Centos' repositories with the following command, replacing `$IP` with the cluster-accessible IP you would like etcd to bind to:
+
+```
+yum install -y etcd
+echo ETCD_LISTEN_CLIENT_URLS=\"http://0.0.0.0:2379\" >> /etc/etcd/etcd.conf
+echo ETCD_ADVERTISE_CLIENT_URLS=\"http://$IP:2379\" >> /etc/etcd/etcd.conf
+systemctl start etcd.service
+```
+
+Etcd listens for tcp connections on port 2379. If you're using a firewall,
+open this port:
+
+```
+sudo firewall-cmd --zone=public --add-port=2379/tcp --permanent
+sudo systemctl restart firewalld
+```
+
+## Mesos Agents
```

Comment:
Yes.

At the very least, we need the **All Agent Types** section.  We explain how to install mesos on a Master, but we don't explain how to install it on an Agent anywhere else.

It is debatable whether we want to include Containerizer info here.  I'd be happy removing it.